Milvus.Client/MilvusCollection.Entity.cs

```diff
@@ -418,61 +418,144 @@ public Task<FlushResult> FlushAsync(CancellationToken cancellationToken = defaul
             Expr = expression
         };
 
-        if (parameters is not null)
+        PopulateQueryRequestFromParameters(request, parameters);
+
+        var response = await _client.InvokeAsync(
+                _client.GrpcClient.QueryAsync,
+                request,
+                static r => r.Status,
+                cancellationToken)
+                .ConfigureAwait(false);
+
+        return ProcessReturnedFieldData(response.FieldsData);
+    }
+
+    /// <summary>
+    /// Retrieves rows from a collection via scalar filtering based on a boolean expression using iterator.
+    /// </summary>
+    /// <param name="expression">A boolean expression determining which rows are to be returned.</param>
+    /// <param name="batchSize">Batch size that will be used for every iteration request. Must be between 1 and 16384.</param>
+    /// <param name="parameters">Various additional optional parameters to configure the query.</param>
+    /// <param name="cancellationToken">
+    /// The token to monitor for cancellation requests. The default value is <see cref="CancellationToken.None" />.
+    /// </param>
+    /// <returns>A list of <see cref="FieldData{TData}" /> instances with the query results.</returns>
+    public async IAsyncEnumerable<IReadOnlyList<FieldData>> QueryWithIteratorAsync(
+        string? expression = null,
+        int batchSize = 1000,
+        QueryParameters? parameters = null,
+        [EnumeratorCancellation] CancellationToken cancellationToken = default)
+    {
+        if ((parameters?.Offset ?? 0) != 0)
         {
-            if (parameters.TimeTravelTimestamp is not null)
+            throw new MilvusException("Not support offset when searching iteration");
+        }
+
+        var describeResponse = await _client.InvokeAsync(
+                _client.GrpcClient.DescribeCollectionAsync,
+                new DescribeCollectionRequest { CollectionName = Name },
+                r => r.Status,
+                cancellationToken)
+                .ConfigureAwait(false);
+
+        var pkField = describeResponse.Schema.Fields.FirstOrDefault(x => x.IsPrimaryKey);
+        if (pkField == null)
+        {
+            throw new MilvusException("Schema must contain pk field");
+        }
+
+        var isUserRequestPkField = parameters?.OutputFieldsInternal?.Contains(pkField.Name) ?? false;
```

Comment:
nit: the project convention is to not use var

Reply:
var used in other places so did not realize this. Will change.